test/service_instance_test/service_instance_test.go

```diff
@@ -1139,36 +1159,35 @@ var _ = DescribeTestsFor(TestCase{
 
 							When("provision responds with error due to time out", func() {
 								var doneChannel chan interface{}
-								var oldCtx *TestContext
+								var newCtx *TestContext
 
 								BeforeEach(func() {
-									oldCtx = ctx
 									doneChannel = make(chan interface{})
-									ctx = NewTestContextBuilderWithSecurity().WithEnvPreExtensions(func(set *pflag.FlagSet) {
-										Expect(set.Set("httpclient.response_header_timeout", (1 * time.Second).String())).ToNot(HaveOccurred())
+									newCtx = t.ContextBuilder.WithEnvPreExtensions(func(set *pflag.FlagSet) {
+										Expect(set.Set("httpclient.response_header_timeout", (2 * time.Second).String())).ToNot(HaveOccurred())
 									}).BuildWithoutCleanup()
 
 									brokerServer.ServiceInstanceHandlerFunc(http.MethodPut, http.MethodPut+"1", DelayingHandler(doneChannel))
 									brokerServer.ServiceInstanceHandlerFunc(http.MethodDelete, http.MethodDelete+"1", ParameterizedHandler(http.StatusOK, Object{}))
 								})
 
 								AfterEach(func() {
-									ctx = oldCtx
+									ctx.SMWithBasic = newCtx.SMWithBasic
```

Comment:
I think it is better to revert to the oldCtx, because with this approach we don't know what exactly to revert and if in future we have to revert more thing than the SMWithBasic it will be hard to manage

Reply:
we need the new basic creds because the test platform record changed (the old ctx basic creds dont work)

Reply:
there are plenty of tests across that semantically should be in the current Context/Describe that they are, but logically need to reconfigure SM and need to bootstrap a new sm. I prefer to keep the BDD/semantic grouping at the cost of the side effect. The side effect can be removed if we iterate over the test context building but this is out of scope ( i think )